OptimizelySDK/OptimizelyUserContext.cs

```diff
@@ -128,6 +143,33 @@ public bool IsQualifiedFor(string segment)
             }
         }
 
+        /// <summary>
+        /// Fetch all qualified segments for the user context.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="callback">Callback function to invoke when results are available</param>
+        /// <param name="segmentOptions">Options used during segment cache handling</param>
+        /// <returns>True if ODP segments were fetched successfully otherwise False</returns>
+        public bool FetchQualifiedSegments(string userId,
+            Action<bool> callback = null,
+            List<OdpSegmentOption> segmentOptions = null
+        )
+        {
+            var segments = Optimizely.FetchQualifiedSegments(userId,
```

Comment:
This should be non-blocking if callback is provided (async API).
We can consider splitting the API into 2 (sync and async).
- bool FetchQualifiedSegments(userId, options)
- void FetchQaulifiedSegments(userId, options, callback)